src/vector_windows.rs

```diff
@@ -0,0 +1,229 @@
+            match s {
+                "install" => Ok(ControlAction::Install),
+                "uninstall" => Ok(ControlAction::Uninstall),
+                "start" => Ok(ControlAction::Start),
+                "stop" => Ok(ControlAction::Stop),
+                _ => Err(format!("invalid option {} for ControlAction", s)),
+            }
+        }
+    }
+
+    pub fn control(service: &ServiceDefinition, action: ControlAction) -> Result<()> {
+        match action {
+            ControlAction::Start => start_service(&service),
+            ControlAction::Stop => stop_service(&service),
+            ControlAction::Install => install_service(&service),
+            ControlAction::Uninstall => uninstall_service(&service),
+        }
+    }
+
+    fn start_service(service: &ServiceDefinition) -> Result<()> {
+        let service_access = ServiceAccess::QUERY_STATUS | ServiceAccess::START;
+        let service = open_service(&service, service_access)?;
+        let service_status = service.query_status()?;
+
+        if service_status.current_state != ServiceState::StartPending
+            || service_status.current_state != ServiceState::Running
+        {
+            service.start(&[] as &[OsString])?;
+        }
+
+        Ok(())
+    }
+
+    fn stop_service(service: &ServiceDefinition) -> Result<()> {
+        let service_access = ServiceAccess::QUERY_STATUS | ServiceAccess::STOP;
+        let service = open_service(&service, service_access)?;
+        let service_status = service.query_status()?;
+
+        if service_status.current_state != ServiceState::StopPending
+            || service_status.current_state != ServiceState::Stopped
+        {
+            service.stop()?;
+        }
+
+        Ok(())
+    }
+
+    fn install_service(service: &ServiceDefinition) -> Result<()> {
+        let manager_access = ServiceManagerAccess::CONNECT | ServiceManagerAccess::CREATE_SERVICE;
+        let service_manager = ServiceManager::local_computer(None::<&str>, manager_access)?;
+
+        let service_info = ServiceInfo {
+            name: service.name.clone(),
+            display_name: service.display_name.clone(),
+            service_type: SERVICE_TYPE,
+            start_type: ServiceStartType::OnDemand,
+            error_control: ServiceErrorControl::Normal,
+            executable_path: service.executable_path.clone(),
+            launch_arguments: service.launch_arguments.clone(),
+            dependencies: vec![],
+            account_name: None,
+            account_password: None,
+        };
+
+        debug!(
+            message = "Registering service.",
+            service_info = format!("{:?}", service_info).as_str(),
+        );
+
+        service_manager.create_service(service_info, ServiceAccess::empty())?;
+        // TODO: It is currently not possible to change the description of the service.
+        // Waiting for the following PR to get merged in
+        // https://github.com/mullvad/windows-service-rs/pull/32
```

Comment:
I see why https://github.com/timberio/vector/pull/2896/files#r474094595 is happening! Ok. :) Thank you. This is fine.